### PR TITLE
[RFC] vim-patch:7.4.{1952, 1990, 2033, 2284}

### LINKED ISSUE #5247: cscope does not support the `a` option
- `nvim --version`: NVIM 0.1.5
- Vim (7.4p2235) behaves differently? Yes
- Operating system/version: OS X 10.11
- Terminal name/version: Terminal.app
- `$TERM`: xterm-256color

### Actual behaviour

NeoVim raises error `E561` when `cscope find a ...` is used. It raises error `E474` when `cscopequickfix` contains `a-`. In short, Cscope's “find assignments” feature is not supported.

### Expected behaviour

`cscope find a ...` should work; `cscopequickfix` should accept `a-`, as Vim does.

### Steps to reproduce using `nvim -u NORC`

```
nvim -u NORC
:set cscopequickfix=a-
:cscope add /path/to/cscope.out
:cscope find a someVar
```

### PR DESCRIPTION
**Pull request by KillTheMule**
Closes https://github.com/neovim/neovim/issues/5247. Also added the documentation change from https://github.com/vim/vim/commit/802a0d902fca423acb15f835d7b09183883d79a0.
## vim-patch:7.4.1952

> Problem:    Cscope interface does not support finding assignments.
> Solution:   Add the "a" command. (ppettina, closes vim/vim#882)

vim/vim@b12e7ef

All changes applied manually.
## vim-patch:7.4.1990

> Problem:    Cscope items are not sorted.
> Solution:   Put the new "a" command first. (Ken Takata)

vim/vim@80632db

All changes applied manually.
## vim-patch:7.4.2033

> Problem:    'cscopequickfix' option does not accept new value "a".
> Solution:   Adjust list of command characters. (Ken Takata)

vim/vim@6d20e17

All changes applied manually. Definition of `CSQF_CMDS` was moved to
`option_defs.h` in nvim, so changes applies there instead of `options.h`.
## vim-patch:7.4.2284

> Problem:    Comment in scope header file is outdated. (KillTheMule)
> Solution:   Point to the help instead. (closes #1017)

https://github.com/vim/vim/commit/f4145d8e990a72bdfea9db3110a7e42a0ff4240c

### CODE REVIEW

File: src/nvim/if_cscope.c [hunk 1]
```diff
@@ -970,6 +970,9 @@ static int cs_find_common(char *opt, char *pat, int forceit, int verbose,
   case '8':
     cmdletter = 'i';
     break;
+  case '9':
+    cmdletter = 'a';
+    break;
```
Comment (justinmk) [inline src/nvim/if_cscope.c]: in the original patch this case was also added to `cs_create_cmd`

**Comment by oni-link** [comment]
In vim/vim@802a0d902fca423acb15f835d7b09183883d79a0 the documentation for `if_cscop.txt` is updated.

**Comment by KillTheMule** [comment]
That's a runtime update, and we're very far from merging it. I propose I merge the relevant doc changes now, the one to port the runtime patch will notice and easily work around.

File: src/nvim/if_cscope.c (hunk 1 above)
Comment (KillTheMule) [inline src/nvim/if_cscope.c]: Wooops, sorry for that, fixed!

**Comment by oni-link** [comment]
A comment block could also be updated [here](https://github.com/neovim/neovim/blob/master/src/nvim/if_cscope_defs.h#L38-Lundefined).

**Comment by KillTheMule** [comment]
Good catch, I updated an opend a [vim issue](https://github.com/vim/vim/issues/1017) so they can fix it as well.

**Comment by justinmk** [comment]
I guess https://github.com/vim/vim/commit/f4145d8e990a72bdfea9db3110a7e42a0ff4240c should be included instead of the comment update.

**Comment by KillTheMule** [comment]
Thanks, justinmk, I included that patch.

**Comment by KillTheMule** [comment]
CI Failures are from master (why doesn't QB fail? It _does_ run the test).

File: src/nvim/version.c [hunk 2]
```diff
@@ -75,6 +75,7 @@ static char *features[] = {
 
 // clang-format off
 static int included_patches[] = {
+  2284,
   2219,
```
Comment (oni-link) [inline src/nvim/version.c]: We are missing some patch number comments from 2200 to 2284.

**Comment by fwalch** [comment]
> CI Failures are from master (why doesn't QB fail? It does run the test).

Travis seems to run https://github.com/neovim/neovim/commit/b6a360fb4037275f3eb572c25f366ef06afef95d, whereas QB ran https://github.com/neovim/neovim/commit/89a7a4f02c35e414bc0bd33560738c9cf0abe33e

**Pull request merged**